samples/cpp/benchmark_app/main.cpp

```diff
@@ -125,43 +125,52 @@ void next_step(const std::string additional_info = "") {
-    ov::hint::PerformanceMode ov_perf_hint = ov::hint::PerformanceMode::UNDEFINED;
-    OPENVINO_SUPPRESS_DEPRECATED_END
+void handle_performance_hint(const std::string& device, const ov::Core& core, ov::AnyMap& config) {
+    ov::hint::PerformanceMode ov_perf_hint = ov::hint::PerformanceMode::THROUGHPUT;
     auto supported_properties = core.get_property(device, ov::supported_properties);
     if (std::find(supported_properties.begin(), supported_properties.end(), ov::hint::performance_mode) !=
         supported_properties.end()) {
-        if (FLAGS_hint != "") {
+        // Use FLAGS_hint to decide performance mode:
+        //
+        // "throughput" or "tput": THROUGHPUT mode
+        // "cumulative_throughput" or "ctput": CUMULATIVE_THROUGHPUT mode
+        // "latency": LATENCY mode
+        // "none": not set ov::hint::performance_mode, let plugin use its default performance mode
+        // ""    : use default THROUGHPUT mode, if FLAG_api="sync" then set LATENCY mode
+        if (FLAGS_hint != "" && FLAGS_hint != "none") {
             if (FLAGS_hint == "throughput" || FLAGS_hint == "tput") {
                 ov_perf_hint = ov::hint::PerformanceMode::THROUGHPUT;
             } else if (FLAGS_hint == "latency") {
                 ov_perf_hint = ov::hint::PerformanceMode::LATENCY;
             } else if (FLAGS_hint == "cumulative_throughput" || FLAGS_hint == "ctput") {
                 ov_perf_hint = ov::hint::PerformanceMode::CUMULATIVE_THROUGHPUT;
-            } else if (FLAGS_hint == "none") {
-                OPENVINO_SUPPRESS_DEPRECATED_START
-                ov_perf_hint = ov::hint::PerformanceMode::UNDEFINED;
-                OPENVINO_SUPPRESS_DEPRECATED_END
             } else {
                 throw std::logic_error(
                     "Incorrect performance hint. Please set -hint option to"
                     "`throughput`(tput), `latency', 'cumulative_throughput'(ctput) value or 'none'.");
             }
-        } else {
-            ov_perf_hint =
-                FLAGS_api == "async" ? ov::hint::PerformanceMode::THROUGHPUT : ov::hint::PerformanceMode::LATENCY;
-
+        } else if (FLAGS_hint == "") {
+            ov_perf_hint = ov::hint::PerformanceMode::THROUGHPUT;
+            if (FLAGS_api == "sync") {
+                ov_perf_hint = ov::hint::PerformanceMode::LATENCY;
+            }
             slog::warn << "Performance hint was not explicitly specified in command line. "
                           "Device("
                        << device << ") performance hint will be set to " << ov_perf_hint << "." << slog::endl;
         }
+
+        if (FLAGS_hint != "none") {
+            // apply command line hint setting and override if hint exists
+            config[ov::hint::performance_mode.name()] = ov_perf_hint;
+        } else {
+            config.erase(ov::hint::performance_mode.name());
+        }
     } else {
-        if (FLAGS_hint != "") {
+        if (FLAGS_hint != "none") {
```

Comment:
Default FLAGS_hint is `""`. Why we change to `"none"` there?

Reply:
As I understand, the "-hint" logic is:
        -hint =  "throughput" or "tput": THROUGHPUT mode
        -hint =  "cumulative_throughput" or "ctput": CUMULATIVE_THROUGHPUT mode
        -hint =  "latency": LATENCY mode
        -hint =  "none":  not set ov::hint::performance_mode, let plugin use its default performance mode
        -hint = "" : use default THROUGHPUT mode, if FLAG_api="sync" then set LATENCY mode

This branch is for device doesn't support `ov::hint::performance_mode` property, and if `hint != "none"` it will report warning. While if `hint == "none"`  it should not pass `ov::hint::performance_mode` to device so need not report warning.